packages/grid/_modules_/grid/utils/warning.ts

```diff
@@ -0,0 +1,12 @@
+export const buildWarning = (message: string | string[]) => {
```

Comment:
In `wrapWithDeprecationWarning` I am doing the env check inside the method, which I think makes the product-code more readable.

And I propose we move the `wrapWithDeprecationWarning` in this file as well and rename it `wrapWithWarning` as there goal is very similar.

Reply:
You can do so because you don't have a if condition to verify before sending the warning.

Since it is a deprecated function, as soon as the function is called you log the message saying "hey it's deprecated"

Here it s not the case, I have a custome `if` between the env check and the warning. So the wrapper will need to get a function computing the condition in which warning must be logged a bit like that, but not sure it makes the code clearer

```js
export const buildConsistencyWarning = (message: string | string[], warningCondition: () => boolean) => {
  if (process.env.NODE_ENV === 'production') {
    return () => null
  }
  let alreadyWarned = false;

  const cleanMessage = Array.isArray(message) ? message.join('\n') : message;

  return (...params) => {
    if (!alreadyWarned && warningCondition(...params)) {
      alreadyWarned = true;
      console.warn(cleanMessage);
    }
  };
};

```

Reply:
It has the benefits of hiding the `env` topic in a single file and be sure to never forget it.
But as you like, I agree that the gain is less clear that for the deprecation for the reason you explaied.